### PR TITLE
makes it so on to spawn, register tiles remove themselves from locations hopefully fixing the null reference bug?

### PR DESCRIPTION
**Pull request by Bod9001**
don't worry the changes have already been merged showing them for some reason though they've already been merged,
also improved explosion code a bit

fixes NRE with table interactions while it's under construction if you don't have anything in your hand
so, mirror with  objects that were in the scene when it was safe when server requests to destroy them it just hides them instead resulting in inconsistent destruction path , (if it were responding then it gets destroyed normally if it was there already it would gethidden) so, I've made it so it destroys even if it was a scene object and tested with a few round restarts with editor hosting and build as client

also removes the annoying " scene needing to be received" thing that stops builds from proceeding,
 for testing purposes becauseit's particularly annoying when it gets half way through a build and stops

### CODE REVIEW

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs [hunk 1]
```diff
@@ -305,6 +305,12 @@ public void OnDestroy()
 
 	public virtual void OnDespawnServer(DespawnInfo info)
 	{
+		if (objectLayer)
+		{
+			objectLayer.ServerObjects.Remove(LocalPositionServer, this);
+			objectLayer.ClientObjects.Remove(LocalPositionClient, this);
+		}
+
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs:313]: Only called on server, wont affect clients

**Comment by ThatDan123** [comment]
Also are all those files in this PR wanted?

**Comment by Bod9001** [comment]
yeah, should make it so walls absorbed more explosion damage

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs:313]: will look into more, dam I really hate that need to save scenes cancel build thing

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterTile.cs:313]: isn't the problem server side?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/TilemapDamage.cs [hunk 2]
```diff
@@ -74,28 +74,44 @@ private float DealDamageAt(float damage, AttackType attackType, Vector3Int cellP
...
-			if(damage >= 1)
+			if(damageTaken >= 1)
 				SoundManager.PlayNetworkedAtPos(basicTile.SoundOnHit, worldPosition);
 		}
 
 		var totalDamageTaken = data.GetTileDamage(Layer.LayerType);
 
 		if (totalDamageTaken >= basicTile.MaxHealth)
 		{
+			float ExcessEnergy = basicTile.Armor.GetForce( totalDamageTaken - basicTile.MaxHealth, attackType);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/TilemapDamage.cs]: camelCase

**Comment by corp-0** [comment]
I have no idea what this PR is for. Is it rebalancing explosions? Is it fixing a bug on RegisterTile like the tittle suggest? What bug is it fixing?

**Comment by corp-0** [comment]
huh, so I guess I should read PR descriptions before reading code changes. Sorry for that, Bod.

**Comment by corp-0** [comment]
Adding a DNM since Bod said he will look into it further.

**Comment by Bod9001** [comment]
Merging so will Be in next build

**Pull request merged**